cyral/resource_cyral_sidecar_listener.go

```diff
@@ -0,0 +1,363 @@
+// POST {{baseURL}}/sidecars/:sidecarID/listeners/ (Create a listener)
+// PUT {{baseURL}}/sidecars/:sidecarID/listeners/:listenerID (Update a listener)
+// DELETE {{baseURL}}/sidecars/:sidecarID/listeners/:listenerID (Delete a listener)
+func resourceSidecarListener() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manages [sidecar listeners](https://cyral.com/docs/sidecars/sidecar-listeners).",
+		CreateContext: CreateResource(
+			ResourceOperationConfig{
+				Name:       "SidecarListenersResourceCreate",
+				HttpMethod: http.MethodPost,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					return fmt.Sprintf("https://%s/v1/sidecars/%s/listeners",
+						c.ControlPlane,
+						d.Get(SidecarIdKey).(string))
+
+				},
+				NewResourceData: func() ResourceData { return &SidecarListenerResource{} },
+				NewResponseData: func(_ *schema.ResourceData) ResponseData { return &CreateListenerAPIResponse{} },
+			}, ReadSidecarListenersConfig,
+		),
+		ReadContext: ReadResource(ReadSidecarListenersConfig),
+		UpdateContext: UpdateResource(
+			ResourceOperationConfig{
+				Name:       "SidecarListenersResourceUpdate",
+				HttpMethod: http.MethodPut,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					return fmt.Sprintf("https://%s/v1/sidecars/%s/listeners/%s",
+						c.ControlPlane,
+						d.Get(SidecarIdKey).(string),
+						d.Get(ListenerIdKey).(string))
+
+				},
+				NewResourceData: func() ResourceData { return &SidecarListenerResource{} },
+			}, ReadSidecarListenersConfig,
+		),
+		DeleteContext: DeleteResource(
+			ResourceOperationConfig{
+				Name:       "SidecarListenersResourceDelete",
+				HttpMethod: http.MethodDelete,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					return fmt.Sprintf("https://%s/v1/sidecars/%s/listeners/%s",
+						c.ControlPlane,
+						d.Get(SidecarIdKey).(string),
+						d.Get(ListenerIdKey).(string))
+				},
+			},
+		),
+		Schema: map[string]*schema.Schema{
+			ListenerIdKey: {
+				Description: "ID of the listener that will be bound to the sidecar.",
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Computed:    true,
+			},
+			SidecarIdKey: {
+				Description: "ID of the sidecar that the listener will be bound to.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			RepoTypesKey: {
+				Description: "List of repository types that the listener supports. Currently limited to one repo type from supported repo types:" + supportedTypesMarkdown(repositoryTypes()),
+				Type:        schema.TypeList,
+				Required:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			NetworkAddressKey: {
+				Description:  "The network address that the sidecar listens on.",
+				Type:         schema.TypeSet,
+				Optional:     true, // This field is required, due to ExactlyOneOf directive
+				ExactlyOneOf: []string{NetworkAddressKey},
```

Comment:
Since now we only have the `NetworkAddress` we can just make this field `required` and remove the `ExactlyOneOf` clause.